homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,170 @@
+    LOGGER,
+    PSS_DEVICE_TYPE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add switch."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        for device in shc_devices:
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in coordinator.switch_devices:
+                    livisi_switch: SwitchEntity = create_entity(
+                        config_entry, device, coordinator
+                    )
+                    LOGGER.debug("Include device type: %s", device.get("type"))
+                    coordinator.switch_devices.append(device)
+                    async_add_entities([livisi_switch])
+
+    coordinator.async_add_listener(handle_coordinator_update)
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> SwitchEntity:
+    """Create Switch Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    room_id: str = device["location"].removeprefix("/location/")
+    room_name: str = coordinator.rooms[room_id]
+    livisi_switch = LivisiSwitch(
+        config_entry,
+        coordinator,
+        unique_id=device["id"],
+        manufacturer=device["manufacturer"],
+        product=device["product"],
+        serial_number=device["serialNumber"],
+        device_type=device["type"],
+        name=config_details["name"],
+        capability_id=capabilities[0],
+        room=room_name,
+    )
+    return livisi_switch
+
+
+class LivisiSwitch(CoordinatorEntity[LivisiDataUpdateCoordinator], SwitchEntity):
+    """Represents the Livisi Switch."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id,
+        manufacturer,
+        product,
+        serial_number,
+        device_type,
+        name,
+        capability_id,
+        room,
+        version=None,
+    ):
+        """Initialize the Livisi Switch."""
+        self.config_entry = config_entry
+        self._attr_unique_id = unique_id
+        self._manufacturer = manufacturer
+        self._product = product
+        self._serial_number = serial_number
+        self._attr_model = device_type
+        self._attr_name = name
+        self._state = None
+        self._capability_id = capability_id
+        self._room = room
+        self._version = version
+        self.aio_livisi = coordinator.aiolivisi
+        self._attr_is_available = False
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self._attr_unique_id))},
```

Comment:
```suggestion
            identifiers={(DOMAIN, unique_id)},
```

Reply:
Now the integration is using `unique_id`.